mirascope/integrations/logfire/_utils.py

```diff
@@ -0,0 +1,180 @@
+"""Mirascope x Logfire Integration utils"""
+
+from contextlib import contextmanager
+from typing import (
+    Any,
+    Awaitable,
+    Callable,
+    Generator,
+    ParamSpec,
+    TypeVar,
+)
+
+import logfire
+from pydantic import BaseModel
+
+from ...core.base import BaseCallResponse
+from ...core.base._stream import BaseStream
+from ...core.base._structured_stream import BaseStructuredStream
+
+_P = ParamSpec("_P")
+_R = TypeVar("_R", bound=BaseCallResponse | BaseStream | BaseModel)
+
+
+@contextmanager
+def custom_context_manager(
+    fn: Callable[_P, _R] | Callable[_P, Awaitable[_R]],
```

Comment:
`_P` and `_R` seem unnecessary here? Should be sufficient to write `Callable` or `Callable[..., Any]` right?

Reply:
done